task.go

```diff
@@ -146,12 +151,14 @@ func (t *task) Status(ctx context.Context) (Status, error) {
 		ContainerID: t.id,
 	})
 	if err != nil {
-		return "", errdefs.FromGRPC(err)
+		return Status{}, errdefs.FromGRPC(err)
 	}
-	return Status(strings.ToLower(r.Process.Status.String())), nil
+	return Status{
+		Status:     ExecutionStatus(strings.ToLower(r.Process.Status.String())),
+		ExitStatus: r.Process.ExitStatus,
+	}, nil
 }
 
-// Wait is a blocking call that will wait for the task to exit and return the exit status
 func (t *task) Wait(ctx context.Context) (uint32, error) {
 	eventstream, err := t.client.EventService().Subscribe(ctx, &eventsapi.SubscribeRequest{
```

Comment:
Ok, this looks good:

1. Cover events.
2. Check that status isn't stopped.
3. Wait events.

@Random-Liu Seems to meet your outline for use!